frontend/src/components/common/CreateResourceButton.tsx

```diff
@@ -0,0 +1,142 @@
+        values.forEach((value: any, index: number) => {
+          if (value.status === 'rejected') {
+            let msg;
+            const kind = newItems[index].kind;
+            const name = newItems[index].metadata.name;
+            const apiVersion = newItems[index].apiVersion;
+            if (newItems.length === 1) {
+              msg = t('translation|Failed to create {{ kind }} {{ name }}.', { kind, name });
+            } else {
+              msg = t('translation|Failed to create {{ kind }} {{ name }} in {{ apiVersion }}.', {
+                kind,
+                name,
+                apiVersion,
+              });
+            }
+            setErrorMessage(msg);
+            setOpenDialog(true);
+            throw msg;
+          }
+        });
+      }
+    );
+  };
+
+  function handleSave(newItemDefs: KubeObjectInterface[]) {
+    let massagedNewItemDefs = newItemDefs;
+    const cancelUrl = location.pathname;
+
+    // check if all yaml objects are valid
+    for (let i = 0; i < massagedNewItemDefs.length; i++) {
+      if (massagedNewItemDefs[i].kind === 'List') {
+        // flatten this List kind with the items that it has which is a list of valid k8s resources
+        const deletedItem = massagedNewItemDefs.splice(i, 1);
+        massagedNewItemDefs = massagedNewItemDefs.concat(deletedItem[0].items);
+      }
+      if (!massagedNewItemDefs[i].metadata?.name) {
+        setErrorMessage(
+          t(`translation|Invalid: One or more of resources doesn't have a name property`)
+        );
+        return;
+      }
+      if (!massagedNewItemDefs[i].kind) {
+        setErrorMessage(t('translation|Invalid: Please set a kind to the resource'));
+        return;
+      }
+    }
+    // all resources name
+    const resourceNames = massagedNewItemDefs.map(newItemDef => newItemDef.metadata.name);
+    setOpenDialog(false);
+
+    const clusterName = getCluster() || '';
+
+    dispatch(
+      clusterAction(() => applyFunc(massagedNewItemDefs, clusterName), {
+        startMessage: t('translation|Applying {{ newItemName }}…', {
+          newItemName: resourceNames.join(','),
+        }),
+        cancelledMessage: t('translation|Cancelled applying {{ newItemName }}.', {
+          newItemName: resourceNames.join(','),
+        }),
+        successMessage: t('translation|Applied {{ newItemName }}.', {
+          newItemName: resourceNames.join(','),
+        }),
+        errorMessage: t('translation|Failed to apply {{ newItemName }}.', {
+          newItemName: resourceNames.join(','),
+        }),
+        cancelUrl,
+      })
+    );
+
+    dispatchCreateEvent({
+      status: EventStatus.CONFIRMED,
+    });
+  }
+
+  function getCurrentResource(resourceName: string) {
+    switch (resourceName) {
+      case 'ConfigMap':
+        return ConfigMap;
+      case 'Lease':
+        return Lease;
+      case 'RuntimeClass':
+        return RuntimeClass;
+      case 'Secret':
+        return Secret;
+    }
+  }
```

Comment:
Please check my comment regarding passing only the class here and checking whether the class has a baseObject defined.